### DISCUSSION TITLE
OnKill status effect

### DISCUSSION THREAD

**Discussion by SquidingTin** [FakeFishGames/Barotrauma#12560] [Ideas and suggestions]
Allowing us to detect when you kill a creature could be very useful, one possible way would be for a melee weapon to change its appearance when killing a creature, or having the ability to heal yourself on kill, health on kill could be very cool from mods if we had the ability to detect kills (current possible way would be with talents, which isnt very easy to deal with)

**Comment by jigardi** [comment]
I think this might potentially synergize very well with talents. It could potentially also be really useful for creating additional gamemodes.
